### ISSUE TITLE
[BUG]: rss guard can't find external web browser under wayland session

### ISSUE THREAD

**Issue by vroger11** [martinrotter/rssguard#750]
### Brief description of the issue

I am using chromium as web browser and rss guard can found it under X session, but not under wayland session

### How to reproduce the bug?

1. be on wayland session
2. try to open a feed why external browser

### What was the expected result?

the external browser open the link

### What actually happened?

error message saying: "RSS Guard was unable to launch your web browser with the given URL, " and so on

### Debug log

I have bo debug log to provide

### Operating system and version

* OS: manjaro KDE
* plasma: 5.24.5
* graphics platform: wayland
* RSS Guard version: flatpak 4.2.3

**Comment by martinrotter** [comment]
I see. Let me check later.

**Comment by martinrotter** [comment]
Re-test with latest dev. build.

**Comment by akinokonomi** [comment]
@vroger11, Do you have problems only when Chromium is set as default system-wide, or did you also try to set custom web browser in RSS Guard settings?

Not sure whether Wayland KDE (or even Wayland Qt) properly support XDG, but it works for me on Sway when I set 
custom web browser in RSS Guard settings (Network & web & tools -> External web browser) to:
```
Web browser executable: /usr/bin/chromium
Parameters: %1
```

Note that URL is corrupt when `%1` is quoted, i.e. `"%1"`, but it still opens that corrupt URL for me without any errors from RSS Guard side.

Tested with `4.2.3` release.

P.s. Did not test Flatpak version, but tested AppImage and native ones.

**Comment by akinokonomi** [comment]
@vroger11 Nevermind. Wayland *does* support XDG, and it works just fine on Sway when I set Chromium as default browser too.
Most likely something to do with Wayland KDE.

**Comment by martinrotter** [comment]
> @vroger11 Nevermind. Wayland _does_ support XDG, and it works just fine on Sway when I set Chromium as default browser too. Most likely something to do with Wayland KDE.

Yes, I briefly tested with my MATE and it works in X11. If it does not work in wayland, it has to be Qt/KDE upstream problem.

Also, "custom browser" opening does not actually use "XDG". It is simple process creation with parameters with [QProcess](https://doc.qt.io/qt-6/qprocess.html). Relatively simple stuff.

**Comment by akinokonomi** [comment]
> If it does not work in wayland

@martinrotter I did all tests on Wayland Sway, and it works for me on both XWayland and native Wayland, as well as with combination of XWayland and native Wayland, exactly as on Xorg.

> "custom browser" opening does not actually use "XDG"

Of course. I only tried to rule out possible causes first. 

The only "default browser" on Linux can be considered the one set in XDG mime,
that's why I asked whether OP has same result when setting Chromium in a non-XDG way. 

But it's irrelevant now,  since both ways work on Sway just fine, which means it has to do with KDE.

**Comment by vroger11** [comment]
Hello, yes I have the same result when setting Chromium in the non-XDG way.
I will look how I can report this problem with KDE devs.
I will let you know if there is any progress on this bug.

**Comment by vroger11** [comment]
Hum, after an update of my flatpaks for kde platform it seems to be working well if I don't set manually the external browser.

So I guess I can make the issue as resolved.

BTW the manual setting should not work with the flatpak session as the app is sandboxed and should not access to all filesystem.

Cheers
